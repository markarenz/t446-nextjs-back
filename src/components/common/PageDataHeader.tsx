import { useState } from 'react';
import styles from '../../styles/modules/PageDataHeader.module.scss';
import IconButton from '../../components/common/IconButton';
import Input from '../../components/common/Input';
import {
  IconNewItem,
  IconPublish,
  IconSearch
} from '../../components/img/icons';

type Props = {
  title: string;
  setSearchFilter: Function;
  defaultSearchText: string;
  handleCreateNew: Function;
  handlePublish: Function;
};

const PageDataHeader: React.FC<Props> = ({
  setSearchFilter,
  defaultSearchText,
  title,
  handleCreateNew,
  handlePublish
}) => {
  const [searchText, setSearchText] = useState(defaultSearchText);
  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearchText(e.target.value);
  };
  const handleSearchClick = () => {
    setSearchFilter(searchText);
  };
  return (
    <div className={styles.pageDataHeader}>
      <h2 className="pageTitle">{title}</h2>
      <div className={styles.controlWrap}>
        <span className="mr-1">
          <Input
            type="search"
            onChange={handleSearchChange}
            autoFocus={false}
            value={searchText}
            onEnterkey={() => handleSearchClick()}
          />
        </span>
        <span className="mr-1">
          <IconButton
            onClick={() => handleSearchClick()}
            title="New Item"
            color="default"
          >
            <IconSearch />
          </IconButton>
        </span>
        <span className="mr-1">
          <IconButton onClick={handlePublish} title="Publish" color="default">
            <IconPublish />
          </IconButton>
        </span>
        <span>
          <IconButton
            onClick={handleCreateNew}
            title="New Item"
            color="primary"
          >
            <IconNewItem />
          </IconButton>
        </span>
      </div>
    </div>
  );
};

export default PageDataHeader;
