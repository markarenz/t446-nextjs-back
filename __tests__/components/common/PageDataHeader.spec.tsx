import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import PageDataHeader from '../../../src/components/common/PageDataHeader';

const defaultProps = {
  title: 'Test',
  defaultSearchText: '',
  setSearchFilter: jest.fn(),
  handleCreateNew: jest.fn(),
  handlePublish: jest.fn()
};
describe('PageDataHeader', () => {
  it('renders component', async () => {
    render(<PageDataHeader {...defaultProps} />);
    const element = await screen.findByTestId('page-data-header');
    expect(element).toBeInTheDocument();
  });
  it('handles search change', async () => {
    render(<PageDataHeader {...defaultProps} />);
    const input = await screen.findByTestId('page-header-search-input');
    const btn = await screen.findByTestId('page-header-search-btn');
    await waitFor(() => fireEvent.change(input, { target: { value: 'new' } }));
    await waitFor(() =>
      fireEvent(
        btn,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true
        })
      )
    );
    expect(defaultProps.setSearchFilter).toHaveBeenCalled();
  });
  it('handles search enter key', async () => {
    render(<PageDataHeader {...defaultProps} />);
    const input = await screen.findByTestId('page-header-search-input');
    await waitFor(() => fireEvent.change(input, { target: { value: 'new' } }));
    fireEvent.keyDown(input, { key: 'Enter', code: 13, charCode: 13 });
    expect(defaultProps.setSearchFilter).toHaveBeenCalled();
  });
});
