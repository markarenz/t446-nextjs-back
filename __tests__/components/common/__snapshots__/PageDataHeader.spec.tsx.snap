// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDataHeader renders component 1`] = `
<div
  class="pageDataHeader"
  data-testid="page-data-header"
>
  <h2
    class="pageTitle"
  >
    Test
  </h2>
  <div
    class="controlWrap"
  >
    <span
      class="mr-1"
    >
      <input
        class="input"
        data-testid="page-header-search-input"
        type="search"
        value=""
      />
    </span>
    <span
      class="mr-1"
    >
      <button
        class="iconButton default"
        data-testid="page-header-search-btn"
        title="New Item"
        type="button"
      >
        <div
          class="inner"
        >
          <svg
            viewBox="0 0 183.792 183.792"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M54.734 9.053C39.12 18.067 27.95 32.624 23.284 50.039c-4.667 17.415-2.271 35.606 6.743 51.22 12.023 20.823 34.441 33.759 58.508 33.759a67.31 67.31 0 0022.287-3.818l30.364 52.592 21.65-12.5-30.359-52.583c10.255-8.774 17.638-20.411 21.207-33.73 4.666-17.415 2.27-35.605-6.744-51.22C134.918 12.936 112.499 0 88.433 0 76.645 0 64.992 3.13 54.734 9.053zm70.556 37.206c5.676 9.831 7.184 21.285 4.246 32.25-2.938 10.965-9.971 20.13-19.802 25.806a42.466 42.466 0 01-21.199 5.703c-15.163 0-29.286-8.146-36.857-21.259-5.676-9.831-7.184-21.284-4.245-32.25 2.938-10.965 9.971-20.13 19.802-25.807A42.47 42.47 0 0188.433 25c15.164 0 29.286 8.146 36.857 21.259z"
            />
          </svg>
        </div>
      </button>
    </span>
    <span
      class="mr-1"
    >
      <button
        class="iconButton default"
        title="Publish"
        type="button"
      >
        <div
          class="inner"
        >
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 11a1 1 0 00-1 1v8c0 1.645 1.355 3 3 3h12c1.645 0 3-1.355 3-3v-8a1 1 0 00-1-1 1 1 0 00-1 1v8c0 .564-.436 1-1 1H6c-.564 0-1-.436-1-1v-8a1 1 0 00-1-1z"
              fill="#000"
            />
            <path
              d="M11.293 1.293l-4 4a1 1 0 000 1.414 1 1 0 001.414 0L12 3.414l3.293 3.293a1 1 0 001.414 0 1 1 0 000-1.414l-4-4a1 1 0 00-1.414 0z"
              fill="#000"
            />
            <path
              d="M12 1a1 1 0 00-1 1v13a1 1 0 001 1 1 1 0 001-1V2a1 1 0 00-1-1z"
              fill="#000"
            />
          </svg>
        </div>
      </button>
    </span>
    <span>
      <button
        class="iconButton primary"
        title="New Item"
        type="button"
      >
        <div
          class="inner"
        >
          <svg
            viewBox="0 0 200 200"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M100 15a85 85 0 1085 85 84.93 84.93 0 00-85-85zm0 150a65 65 0 1165-65 64.87 64.87 0 01-65 65zm30-72.5h-20v-20a10 10 0 00-20 0v20H70a10 10 0 000 20h20v20a10 10 0 0020 0v-20h20a10 10 0 000-20z"
            />
          </svg>
        </div>
      </button>
    </span>
  </div>
</div>
`;
